### PR TITLE
variable binding global fallback (#313)

### PR DESCRIPTION
**Pull request by dinhlongviolin1**
Solve Issue #313

### CODE REVIEW

**Review by FredLL-Avaiga — APPROVED** [review]
Any tests ?

File: taipy/gui/gui.py [hunk 1]
```diff
@@ -1050,9 +1056,19 @@ def _get_partial(self, route: str) -> t.Optional[Partial]:
...
     def _bind_var(self, var_name: str) -> str:
-        encoded_var_name = self.__var_dir.add_var(var_name, self._get_locals_context())
-        if not hasattr(self._bindings(), encoded_var_name) and var_name in self._get_locals_bind().keys():
-            self._bind(encoded_var_name, self._get_locals_bind()[var_name])
+        bind_context = None
+        if var_name in self._get_locals_bind().keys():
+            bind_context = self._get_locals_context()
+        elif var_name not in self._get_default_locals_bind().keys():
+            raise RuntimeError(
+                f"Variable '{var_name}' is not available in both '{self._get_locals_context()}' module and '__main__' module"
```
Comment (FabienLelaquais) [inline taipy/gui/gui.py]: "both" -> " neither"

**Comment by dinhlongviolin1** [comment]
> Any tests ?

Will have tests! I am fixing tests and adding tests.

**Review by FabienLelaquais — APPROVED** [review]
Looking good!

File: taipy/gui/gui.py (hunk 1 above)
Comment (dinhlongviolin1) [reply taipy/gui/gui.py]: thanks!

**Pull request merged**